### PR TITLE
Support tags in detection functions

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This fills in another functionality gap between native AA functions and Nuclio functions.

Note that there is no server counterpart to this PR, because the server-side agent code already works correctly with function results containing tags.

### How has this been tested?

Unit tests.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat-sdk/cvat_sdk/auto_annotation/driver.py [hunk 1]
```diff
@@ -340,15 +345,15 @@ def _remap_attribute(
 
     def _remap_attributes(
         self,
-        shape: Union[models.LabeledShapeRequest, models.SubLabeledShapeRequest],
+        tors: Union[TagOrShape, models.SubLabeledShapeRequest],
```
Comment (zhiltsov-max) [inline cvat-sdk/cvat_sdk/auto_annotation/driver.py]:
```suggestion
        annotation: Union[TagOrShape, models.SubLabeledShapeRequest],
```

File: tests/python/sdk/test_auto_annotation.py [hunk 2]
```diff
@@ -1038,6 +1089,29 @@ def test_attribute_val_unsuitable_for_spec(self):
...
+            super().__init__()
+            self._label_id = label_id
+
+        def forward(self, images: torch.Tensor) -> torch.Tensor:
+            assert isinstance(images, torch.Tensor)
+
+            probs = [0.0] * 1000
+            # Since the function does a softmax operation over the output,
+            # we can't test edge cases by making the final probability exactly 0.75.
+            # With this value the final probability is ~0.77.
```
Comment (zhiltsov-max) [inline tests/python/sdk/test_auto_annotation.py:1103]: I don't really see a point in trying to hit the exact threshold value here. These are floating point values that can be arbitrary. Just saying that this value produces something between 0.75 and 0.9 would be fine.

File: cvat-sdk/cvat_sdk/auto_annotation/functions/torchvision_classification.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import PIL.Image
+
+import cvat_sdk.auto_annotation as cvataa
+import cvat_sdk.models as models
+
+from ._torchvision import TorchvisionFunction
+
+
+class _TorchvisionDetectionFunction(TorchvisionFunction):
```
Comment (zhiltsov-max) [inline cvat-sdk/cvat_sdk/auto_annotation/functions/torchvision_classification.py]:
```suggestion
class _TorchvisionClassificationFunction(TorchvisionFunction):
```
?

File: cvat-cli/src/cvat_cli/_internal/agent.py [hunk 4]
```diff
@@ -802,15 +802,16 @@ def _calculate_result_for_annotate_task_ar(self, ar_id: str, ar_params) -> dict[
 
         mapper = self._create_annotation_mapper_for_detection_ar(ar_params, ds.labels)
 
-        all_annotations = models.PatchedLabeledDataRequest(shapes=[])
+        all_annotations = models.PatchedLabeledDataRequest(tags=[], shapes=[])
 
         for sample_index, sample in enumerate(ds.samples):
             context = self._create_detection_function_context(ar_params, sample.frame_name)
-            shapes = self._executor.result(
+            tags_and_shapes = self._executor.result(
```
Comment (zhiltsov-max) [inline cvat-cli/src/cvat_cli/_internal/agent.py]: Probably need to update the return value type in the line 165

File: cvat-sdk/cvat_sdk/auto_annotation/driver.py (hunk 1 above)
Comment (SpecLad) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: Are you proposing this for everywhere? Or just here?

Comment (zhiltsov-max) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: Ideally to replace tors with something more meaningful everywhere.

Comment (zhiltsov-max) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: I can suggest DetectionAnnotation or DetectorAnnotation, since we call such functions Detectors in CVAT.

Comment (SpecLad) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: Hmm, maybe just `Detection` then?

File: tests/python/sdk/test_auto_annotation.py (hunk 2 above)
Comment (SpecLad) [reply tests/python/sdk/test_auto_annotation.py:1103]: > I don't really see a point in trying to hit the exact threshold value here.

The point would be to test the behavior when the confidence exactly equals the threshold. The tests for the other torchvision functions do this.

> Just saying that this value produces something between 0.75 and 0.9 would be fine.

Sure, but isn't giving the approximate number also fine?

Comment (zhiltsov-max) [reply tests/python/sdk/test_auto_annotation.py:1103]: The code is fine, my comment is only about the line

> we can't test edge cases by making the final probability exactly 0.75.

File: cvat-sdk/cvat_sdk/auto_annotation/driver.py (hunk 1 above)
Comment (zhiltsov-max) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: Tags are not really "detections" in the computer vision sense, so I'd vote against this variant. It might be ok in the current CVAT terminology, but it looks a little bit broken to me at the moment.

Comment (SpecLad) [reply cvat-sdk/cvat_sdk/auto_annotation/driver.py]: Yeah, I agree. I went with `DetectionAnnotation`, which seems to be most consistent with existing code.

File: cvat-sdk/cvat_sdk/auto_annotation/functions/torchvision_classification.py (hunk 3 above)
Comment (SpecLad) [reply cvat-sdk/cvat_sdk/auto_annotation/functions/torchvision_classification.py]: Right, thanks. Fixed.

File: cvat-cli/src/cvat_cli/_internal/agent.py (hunk 4 above)
Comment (SpecLad) [reply cvat-cli/src/cvat_cli/_internal/agent.py]: Updated.

File: tests/python/sdk/test_auto_annotation.py (hunk 2 above)
Comment (SpecLad) [reply tests/python/sdk/test_auto_annotation.py:1103]: Okay. So do you think I should change something?

**Review by zhiltsov-max — APPROVED** [review]

File: tests/python/sdk/test_auto_annotation.py (hunk 2 above)
Comment (zhiltsov-max) [reply tests/python/sdk/test_auto_annotation.py:1103]: I thought to remove the part about the edge cases, because I was confused by the wording, but I don't think it's a stopper anyway.
